sdk/core/src/services/WalletStorageService.ts

```diff
@@ -371,15 +365,137 @@ export default class WalletStorageService {
+    const token = await this._authorizeVcVault(audienceDid)
+    storageRegion = storageRegion || this._storageRegion
+
+    const responses = []
+
+    const encryptedCredentials = await this._encryptCredentials(credentials)
+
+    for (const credential of encryptedCredentials) {
+      const { body } = await this._vaultApiService.storeCredential(token, storageRegion, credential.credentialId, {
+        credentialTypes: credential.credentialTypes,
+        payload: credential.payload,
+      })
+
+      responses.push(body)
+    }
+
+    return responses
+  }
+
+  public async getAllCredentials(audienceDid: string, types: string[][], storageRegion?: string) {
+    const token = await this._authorizeVcVault(audienceDid)
+    storageRegion = storageRegion || this._storageRegion
+    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
+
+    // convert each type to hash
+    const encryptedTypes: string[][] = []
+    for (const subset of types) {
+      const encryptedSubset: string[] = []
+      for (const type of subset) {
+        const encryptedType = await this._platformEncryptionTools.computePersonalHash(privateKeyBuffer, type)
+        encryptedSubset.push(encryptedType)
+      }
+      encryptedTypes.push(encryptedSubset)
+    }
+
+    const { body } = await this._vaultApiService.searchCredentials(token, storageRegion, encryptedTypes)
+    const credentials = await this._decryptCredentials(body.credentials)
+
+    // should be deleted during migration to affinidi-vault Phase #2
+    const bloomCredentials = await this.fetchAllDecryptedCredentials()
+    if (types.length === 0) {
+      credentials.push(...bloomCredentials)
+      return credentials
+    }
+
+    for (const credential of bloomCredentials) {
+      const credentialTypes = Array.isArray(credential?.type) ? (credential.type as string[]) : []
+
+      const isFits = types.some((subtypes) => subtypes.every((subtype) => credentialTypes.includes(subtype)))
+
+      if (isFits) {
+        credentials.push(credential)
+      }
+    }
+
+    return credentials
+  }
+
+  public async getCredentialById(audienceDid: string, credentialId: string, storageRegion?: string): Promise<any> {
+    const token = await this._authorizeVcVault(audienceDid)
+    storageRegion = storageRegion || this._storageRegion
+    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
+
+    const hashedId = await this._platformEncryptionTools.computePersonalHash(privateKeyBuffer, credentialId)
+
+    let encryptedCredential: string
+    try {
+      const { body, status } = await this._vaultApiService.getCredential(token, storageRegion, hashedId)
+      if (status >= 400) {
+        throw new SdkErrorFromCode('COR-23', { credentialId })
+      }
+
+      encryptedCredential = body.payload
+    } catch (error) {
+      // should be deleted during migration to affinidi-vault Phase #2
+      const bloomToken = await this.authorizeVcBloomVault()
```

Comment:
We probably should only do that for 404 error, not just for any exception that was thrown from `getCredential`.

Reply:
> get by credential id: should go to both Bloom Vault and Affinidi Vault returning the first successful response. If both fails, return Affinidi Vault error.

Reply:
that what we have in ADR, so probably make sense do that for every

Reply:
Yes, it looks like the wording in ADR was imprecise.

What it is supposed to mean is that we should try to find a credential with this ID in Affinidi Vault; and go to the Bloom Vault if it is not in the Affinidi Vault. Not that we should try to find it in Bloom Vault if Affinidi Vault returned 500 Internal Server Error, that would make little sense.

The reason for using both Affinidi Vault and Bloom Vault is that some credentials might be stored in Affinidi Vault only, while some others might be stored in Bloom Vault only; and if credential with that id is stored _anywhere_, we should return it no matter where it is stored. Not that Bloom Vault is some kind of a fallback for when Affinidi Vault is unavailable.

Reply:
Okay, but `500 Internal Server Error` in Affinidi Vault doesn't guarantee that credential exists in Affinidi Vault and not in Bloom Vault. So in worst case we will get 500 at affinidi-vault call and 404 (or some other) in bloom-vault, then will throw error from affinidi-vault. But in case credential present in bloom-vault we reject getting it because affinidi-vault is unavailable.

Reply:
That's an overcomplication, Bloom Vault is not intended to be a failsafe for Affinidi Vault.

If Affinidi Vault returned 500 Internal Server Error, the best thing to do is to rethrow it to the caller instead of doing any more processing.

Affinidi Vault should not return 500; and if it does, we don't have to write complex logic to handle it in some way and to fallback to Bloom Vault.

Reply:
Okay, I can handle 404 error only, but all this logic complex left as it, nothing else should be written, except new line:
```
} catch (error) {
      if (error.httpStatusCode === 404) {
```
We still need try/catch as GenericApiService throws an error if status code not starting with `2`, right?

Reply:
Yes, we still need try/catch there.

And also it's probably better to rely on error code than on HTTP status code.

So it could look like:

```
try {
  const { body } = loadFromAffinidiVault(...)
  encryptedCredential = body.payload
} catch (err) {
  if (err.code !== 'AVT-2') {
    throw err
  }

  const blobInfo = loadFromBloomVault(...)
  encryptedCredential = blobInfo.cyphertext
}
```

Or even better, if you separate this into three classes as @Yengas suggested, the entire method could look like:

```
try {
  return loadFromAffinidiVault(id)
} catch (err) {
  if (err.code !== 'AVT-2') {
    throw err
  }

  return loadFromBloomVault(id)
}
```

Reply:
yes, that what I just did. Currently working on splitting logic to separate services